raiden/tests/integration/test_pythonapi.py

```diff
@@ -364,3 +366,44 @@ def test_funds_check_for_openchannel(raiden_network, token_addresses):
     # so that the gas reserve checks cannot pass in parallel
     with pytest.raises(InsufficientGasReserve):
         gevent.joinall(greenlets, raise_error=True)
+
+
+@pytest.mark.parametrize('number_of_nodes', [2])
+@pytest.mark.parametrize('channels_per_node', [1])
+@pytest.mark.parametrize('reveal_timeout', [8])
+@pytest.mark.parametrize('settle_timeout', [30])
+def test_payment_timing_out_if_partner_does_not_respond(
```

Comment:
I also added a unit test for the emitting of the event but I think that an integration test for this scenario is not a bad idea.